app/assets/javascripts/sidebar.js

```diff
@@ -0,0 +1,7 @@
+
+$(document).ready(function() {
+  $('.time-period-select').select2();
+  $( ".time-period-select" ).change(function(event) {
+    window.location.replace(window.location.pathname + `?period=${event.target.value}`)
```

Comment:
this seems a bit to manual, why don't you work with a form like here https://medium.com/swlh/advanced-filtering-for-your-rails-5-application-28c8da2d29b6 ?